integration/setup/teardown.go

```diff
@@ -4,18 +4,15 @@ import (
 	"context"
 
 	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/azure-operator/integration/env"
 )
 
 // Teardown e2e testing environment.
 func Teardown(c Config) error {
 	ctx := context.Background()
 
-	// TODO the deletion detection should rather happen based on the cluster
-	// status or even based on the CR being gone.
-	//
-	//     https://github.com/giantswarm/giantswarm/issues/3839
-	//
-	err := c.Host.DeleteGuestCluster(ctx, "azure")
+	_, err := c.AzureClient.ResourceGroupsClient.Delete(ctx, env.ClusterID())
```

Comment:
> Why don't we delete the resource group and forget about it?

Would it make sense to do both? My concern here is that when we just delete the resource group, is there a chance that `azure-operator` re-creates it when reconciling `AzureConfig` without `deleteTimestamp`?

Reply:
Good point. I've changed the code to first remove the Custom Resource.